File: sample-app/src/debug/java/org/granchi/whatnow/framework/dagger/DebugWhatNowComponent.java
package org.granchi.whatnow.framework.dagger;

import org.granchi.whatnow.DebugWhatNowApplication;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Dagger component for the debug build variant of the app.
 *
 * @author serandel
 */
@Singleton
@Component(modules = {DebugWhatNowModule.class})
public interface DebugWhatNowComponent extends WhatNowComponent {
    void inject(DebugWhatNowApplication app);
}

File: sample-app/src/debug/kotlin/org/granchi/whatnow/framework/dagger/DebugWhatNowComponent.kt
package org.granchi.whatnow.framework.dagger

import dagger.Component
import org.granchi.whatnow.DebugWhatNowApplication
import javax.inject.Singleton

/**
 * Dagger component for the debug build variant of the app.
 */
@Singleton
@Component(modules = arrayOf(DebugWhatNowModule::class))
interface DebugWhatNowComponent : WhatNowComponent {
    fun inject(app: DebugWhatNowApplication)
}
